tests/emscripten_runner.js

```diff
@@ -0,0 +1,39 @@
+const { opendir } = require('node:fs/promises');
+const { loadPyodide } = require('pyodide');
+
+async function findWheel(distDir) {
+  const dir = await opendir(distDir);
+  for await (const dirent of dir) {
+    if (dirent.name.endsWith('whl')) {
+      return dirent.name;
+    }
+  }
+}
+
+const pkgDir = process.argv[2];
+const distDir = pkgDir + '/dist';
+const testDir = pkgDir + '/tests';
+
+async function main() {
+  const wheelName = await findWheel(distDir);
+  const wheelURL = `file:${distDir}/${wheelName}`;
+
+  try {
+    pyodide = await loadPyodide();
+    const FS = pyodide.FS;
+    const NODEFS = FS.filesystems.NODEFS;
+    FS.mkdir('/test_dir');
+    FS.mount(NODEFS, { root: testDir }, '/test_dir');
+    await pyodide.loadPackage(['micropip', 'pytest', 'pytz']);
+    const micropip = pyodide.pyimport('micropip');
+    await micropip.install('dirty-equals');
```

Comment:
What happens if we leave out dirty-equals?

Reply:
```
nox > node emscripten_runner.js /Users/messense/Projects/pydantic-core
Loading distutils
Loaded distutils
Python initialization complete
distutils already loaded from default channel
Loading micropip, pyparsing, packaging, pytest, atomicwrites, attrs, six, more-itertools, pluggy, py, setuptools, iniconfig, pytz
Loaded pyparsing, packaging, micropip, six, attrs, iniconfig, pluggy, pytest, py, atomicwrites, more-itertools, pytz, setuptools
Loading typing-extensions
Loaded typing-extensions
============================= test session starts ==============================
platform emscripten -- Python 3.10.2, pytest-7.1.2, pluggy-1.0.0 --
cachedir: .pytest_cache
rootdir: /test_dir
collecting ...
collecting 497 items / 2 errors
collected 754 items / 4 errors

==================================== ERRORS ====================================
___________________ ERROR collecting validators/test_dict.py ___________________
ImportError while importing test module '/test_dir/validators/test_dict.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/lib/python3.10/site-packages/_pytest/python.py:608: in _importtestmodule
    mod = import_path(self.path, mode=importmode, root=self.config.rootpath)
/lib/python3.10/site-packages/_pytest/pathlib.py:533: in import_path
    importlib.import_module(module_name)
/lib/python3.10/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:1050: in _gcd_import
    ???
<frozen importlib._bootstrap>:1027: in _find_and_load
    ???
<frozen importlib._bootstrap>:1006: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:688: in _load_unlocked
    ???
/lib/python3.10/site-packages/_pytest/assertion/rewrite.py:168: in exec_module
    exec(co, module.__dict__)
/test_dir/validators/test_dict.py:6: in <module>
    from dirty_equals import HasRepr, IsStr
E   ModuleNotFoundError: No module named 'dirty_equals'
___________________ ERROR collecting validators/test_list.py ___________________
ImportError while importing test module '/test_dir/validators/test_list.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/lib/python3.10/site-packages/_pytest/python.py:608: in _importtestmodule
    mod = import_path(self.path, mode=importmode, root=self.config.rootpath)
/lib/python3.10/site-packages/_pytest/pathlib.py:533: in import_path
    importlib.import_module(module_name)
/lib/python3.10/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:1050: in _gcd_import
    ???
<frozen importlib._bootstrap>:1027: in _find_and_load
    ???
<frozen importlib._bootstrap>:1006: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:688: in _load_unlocked
    ???
/lib/python3.10/site-packages/_pytest/assertion/rewrite.py:168: in exec_module
    exec(co, module.__dict__)
/test_dir/validators/test_list.py:4: in <module>
    from dirty_equals import IsList, IsNonNegative
E   ModuleNotFoundError: No module named 'dirty_equals'
__________________ ERROR collecting validators/test_model.py ___________________
ImportError while importing test module '/test_dir/validators/test_model.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/lib/python3.10/site-packages/_pytest/python.py:608: in _importtestmodule
    mod = import_path(self.path, mode=importmode, root=self.config.rootpath)
/lib/python3.10/site-packages/_pytest/pathlib.py:533: in import_path
    importlib.import_module(module_name)
/lib/python3.10/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:1050: in _gcd_import
    ???
<frozen importlib._bootstrap>:1027: in _find_and_load
    ???
<frozen importlib._bootstrap>:1006: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:688: in _load_unlocked
    ???
/lib/python3.10/site-packages/_pytest/assertion/rewrite.py:168: in exec_module
    exec(co, module.__dict__)
/test_dir/validators/test_model.py:8: in <module>
    from dirty_equals import HasRepr, IsStr
E   ModuleNotFoundError: No module named 'dirty_equals'
__________________ ERROR collecting validators/test_tuple.py ___________________
ImportError while importing test module '/test_dir/validators/test_tuple.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/lib/python3.10/site-packages/_pytest/python.py:608: in _importtestmodule
    mod = import_path(self.path, mode=importmode, root=self.config.rootpath)
/lib/python3.10/site-packages/_pytest/pathlib.py:533: in import_path
    importlib.import_module(module_name)
/lib/python3.10/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:1050: in _gcd_import
    ???
<frozen importlib._bootstrap>:1027: in _find_and_load
    ???
<frozen importlib._bootstrap>:1006: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:688: in _load_unlocked
    ???
/lib/python3.10/site-packages/_pytest/assertion/rewrite.py:168: in exec_module
    exec(co, module.__dict__)
/test_dir/validators/test_tuple.py:4: in <module>
    from dirty_equals import IsNonNegative, IsTuple
E   ModuleNotFoundError: No module named 'dirty_equals'
```

Reply:
Ah right of course it's a test dependence. Hmmm @samuelcolvin do you have a `test-requirements.txt` file or something? It would be great to grab the test requirements from the same place.

Reply:
There is https://github.com/samuelcolvin/pydantic-core/blob/main/tests/requirements.txt